src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.cs

```diff
@@ -217,11 +220,15 @@ public static void CompileToAssembly(RegexCompilationInfo[] regexinfos, Assembly
                 ThrowHelper.ThrowArgumentNullException(ExceptionArgument.regexinfos);
             }
 
-#if DEBUG // until it can be fully implemented
-            RegexCompiler.CompileToAssembly(regexinfos, assemblyname, attributes, resourceFile);
-#else
+            foreach (RegexCompilationInfo info in regexinfos)
```

Comment:
is this taking compat to an extreme? someone here is trying to compile, but passing in a null by accident, and will break if they get PNSE instead of ANE?

if that's all it is then my suggestion is to just delete all the validation in this method and throw PNSE. if eg a method is not valid on Unix, we just throw PNSE we dont validate the parameters first.

I don't feel strongly though.

Reply:
I don't have a strong opinion, either, though regarding:

>  eg a method is not valid on Unix, we just throw PNSE we dont validate the parameters first.

in many places we do actually validate arguments before throwing PNSE.

Reply:
I only found one but that was based on very crude greppage. Anyway - either is fine.

Reply:
Any place we have XxCore methods that throw PNSE generally have argument validation performed before that Core method, e.g.
https://github.com/dotnet/runtime/blob/8e9ab9f885644d6dee0b4813982507c10db14bf9/src/libraries/System.Private.CoreLib/src/System/Threading/Semaphore.Unix.cs#L17

Reply:
Ah yes, I thought we were talking about checking for invalid args, not specific sets of args that were not supported. Anyway - my bad.